src/thread.h

```diff
@@ -62,8 +63,8 @@ inline bool StartNewThread(std::thread *thr, const char *name, TFn&& _Fx, TArgs&
 				try {
 					/* Call user function with the given arguments. */
 					F(A...);
-				} catch (...) {
-					NOT_REACHED();
+				} catch (std::exception &e) {
+					FatalError("Unhandled exception in {} thread: {}", name, e.what());
```

Comment:
Not everything that's thrown is a (sub class of) `std::exception`. For example win32_s.cpp throwing C-string pointers, or `OTTDByteReaderSignal` and `AbortGenerateWorldSignal` that are not a sub-class of `std::exception`.

Maybe retaining the `catch (...)` with `NOT_REACHED()` is better over the default `abort()` behaviour? So first catching `std::exception` and if that doesn't match falling back to catching `...`.

Reply:
`OTTDByteReaderSignal` should never reach this point (newgrf code will catch it).
`AbortGenerateWorldSignal` is handle in `_GenerateWorld()` so should not reach this point either.
win32_s.cpp ones are also handled (in `SoundDriver_Win32::Start()`).
But I agree, I can keep the `...` just in case.